### PR TITLE
DHFPROD-2567: To run a custom step, user can't leave the source collection empty

### PR DESCRIPTION
**Pull request by brucean52**
Also truncated step name (DHFPROD-2514)

### CODE REVIEW

File: web/src/main/ui/app/components/flows-new/validators/new-step-dialog.validator.ts [hunk 1]
```diff
@@ -3,19 +3,24 @@ import { StepType } from '../models/step.model';
...
+  switch (group.value.stepDefinitionType) {
+    case StepType.MAPPING:
+    case StepType.MASTERING:
+      if (!group.value.targetEntity) {
+        errors['noTargetEntity'] = true;
+      }
+      if (
+        (group.value.selectedSource === 'collection' && !group.value.sourceCollection) ||
+        (group.value.selectedSource === 'query' && !group.value.sourceQuery) ||
+        (group.value.selectedSource === '')
```
Comment (clockworked247) [inline web/src/main/ui/app/components/flows-new/validators/new-step-dialog.validator.ts]: As Aries mentioned, choosing a selected source isn't required.   But, if they do choose a source type, they must have a sourceCollection value with collection type and sourceQuery with query type.

Comment (clockworked247) [reply web/src/main/ui/app/components/flows-new/validators/new-step-dialog.validator.ts]: so the statement

> (group.value.selectedSource === '')

should be removed

**Comment by brucean52** [comment]
@clockworked247 @wooldridge I removed the source validation requirement for custom step and added logic to allow empty source collection/query. We are relying on the new step validators to not allow mapping and mastering to have empty source collection/query

**Comment by brucean52** [comment]
@ayuwono Good catch. I added the condition for default source query if no radio selection is made

**Comment by ayuwono** [comment]
> @brucean52 if I do the following:
> 
> 1. Create Custom step.
> 2. Select "Query" as the Source Type.
> 3. DO NOT specify a source query and save the step.
> 
> I do not see a (default) sourceQuery value in the Custom step artifact. Should there be?

@wooldridge , I'm seeing this after running your step:
{
  "name": "FlowD",
  "description": "",
  "batchSize": 100,
  "threadCount": 4,
  "stopOnError": false,
  "options": {},
  "version": 0,
  "steps": {
    "1": {
      "name": "CustomD",
      "description": "",
      "options": {
        "sourceQuery": "cts.collectionQuery([])",
        "targetEntity": "",
        "collections": [
          "CustomD"
        ],
        "sourceDatabase": "data-hub-STAGING",
        "sourceCollection": "",
        "targetDatabase": "data-hub-FINAL"
      },
      "customHook": null,
      "retryLimit": null,
      "batchSize": null,
      "threadCount": null,
      "stepDefinitionName": "CustomD",
      "stepDefinitionType": "CUSTOM",
      "modulePath": "/custom-modules/custom/CustomD/main.sjs"
    }
  }
}

**Comment by wooldridge** [comment]
@ayuwono thanks for checking this. I rescind my previous comment as now I see the default value filled in now and the step runs fine. I suspect it was a hiccup in my editor environment.

**Comment by brucean52** [comment]
@ayuwono @wooldridge Thanks for testing this. I updated the truncate in the top stepper container and added truncating to the bottom step container. We can revisit the tooltip option in another JIRA ticket.

**Comment by ayuwono** [comment]
On the bottom step container

At first I see this when my browser window is not extended:
Screen Shot 2019-05-30 at 1 44 16 PM

Then I extend my browser window and the truncate looks fine, but the vertical dots and chevron menu icon are outside the pane:
Screen Shot 2019-05-30 at 1 44 29 PM

**Comment by ayuwono** [comment]
Very close @brucean52 =) , I like having the "..." and the end of the truncate if you can, instead of abruptly cut

Screen Shot 2019-05-30 at 2 44 55 PM

**Comment by brucean52** [comment]
@ayuwono Thanks for all the testing. It should be good now :)

**Pull request merged**